packages/js-moi-wallet/dist/wallet.d.ts

```diff
@@ -1,40 +1,43 @@
 /// <reference types="node" />
 import { Buffer } from "buffer";
-import { Signer, SigType } from "js-moi-signer";
-import { AbstractProvider, InteractionRequest, InteractionObject } from "js-moi-providers";
+import { AbstractProvider, InteractionObject, InteractionRequest } from "js-moi-providers";
+import { SigType, Signer } from "js-moi-signer";
 import { Keystore } from "../types/keystore";
 export declare enum CURVE {
     SECP256K1 = "secp256k1"
 }
 /**
- * Wallet
+ * A class representing a wallet that can sign interactions.
  *
- * A class representing a wallet that can sign interactions and manage keys.
+ * The Wallet implements the Signer API and can be used anywhere a [Signer](https://js-moi-sdk.docs.moi.technology/signer)
+ * is expected and has all the required properties.
+ *
+ * @example
+ * // creating a wallet from mnemonic
+ * const wallet = await Wallet.fromMnemonic("hollow appear story text start mask salt social child ...");
+ *
+ * @example
+ * // creating a wallet from keystore
+ * const keystore = { ... }
+ * const wallet = Wallet.fromKeystore(keystore, "password");
+ *
+ * @example
+ * // Connecting a wallet to a provider
+ * const wallet = await Wallet.fromMnemonic("hollow appear story text start mask salt social child ...");
+ * const provider = new VoyagerProvider("babylon");
+ *
+ * wallet.connect(provider);
+ *
+ * @docs https://js-moi-sdk.docs.moi.technology/hierarchical-deterministic-wallet
  */
 export declare class Wallet extends Signer {
-    constructor(provider?: AbstractProvider);
-    /**
-     * Initializes the wallet with a private key, mnemonic, and curve.
-     *
-     * @param {Buffer} key - The private key as a Buffer.
-     * @param {string} curve - The elliptic curve algorithm used for key generation.
-     * @param {string} mnemonic - The mnemonic associated with the wallet. (optional)
-     * @throws {Error} if the key is undefined or if an error occurs during the
-     * initialization process.
-     */
-    load(key: Buffer, curve: string, mnemonic?: string): void;
+    constructor(key: Buffer, curve: string, mnemonic?: string);
```

Comment:
do not mix both private key and mnemonic. As there is already a static method to create a wallet from a mnemonic, I don't see any need for a mnemonic in the constructor.

Reply:
refactored in: a36b11b60f5391536a65926f9450046806d84589

--- 
- Remove storing of mnemonic in `__vault` in constructor function:

  https://github.com/sarvalabs/js-moi-sdk/blob/8d175a81c59d9b615201a41ac64f4adfee3bec44/packages/js-moi-wallet/src/wallet.ts#L90-L123